asv_bench/benchmarks/tslibs/timestamp.py

```diff
@@ -28,6 +32,12 @@ def time_fromordinal(self):
     def time_fromtimestamp(self):
         Timestamp.fromtimestamp(1515448538)
 
+    def time_identity_scalar(self):
+        Timestamp(self.ts)
+
+    def time_identity_series_apply(self):
+        self.ts_series.apply(lambda x: Timestamp(x))
```

Comment:
@jreback 

Added benchmarks for scalar cast and apply. Didn't touch astype, as that has nothing to do with this PR. Is this what you had in mind?

Reply:
benchmark.tslibs is intended to only depend on _libs.tslibs, so shouldnt have any Series objects in it.  just benchmark the Timestamp contructor here

Reply:
as long as we're at it, might as well have cases for single-arguments for np.datetime64, pydatetime, and tzaware